application/models/Airports.php

```diff
@@ -3,48 +3,26 @@
 /**
  * Dummy model class that represents the airports our airline services
  */
-class Airports extends CI_Model
+class Airports extends CSV_Model
 {
 
-    // The mock airport data
-    var $data = array(
-        'KYPR' => array(
-            'code' => 'YPR',
-            'location' => 'Prince Rupert Airport'),
-        'KZMT' => array(
-            'code' => 'ZMT',
-            'location' => 'Masset Airport'),
-        'KYZP' => array(
-            'code' => 'YZP',
-            'location' => 'Sandspit Airport'),
-        'KYXT' => array(
-            'code' => 'YXT',
-            'location' => 'Terrace Airport'),
-    );
-
     // Constructor
     public function __construct()
     {
-        parent::__construct();
-
-        // inject each "record" key into the record itself, for ease of presentation
-        foreach ($this->data as $key => $record)
-        {
-            $record['key'] = $key;
-            $this->data[$key] = $record;
-        }
+        parent::__construct(APPPATH . DATA_AIRPORTS, 'id');
     }
 
-    // Get a single airplane based off its id, or null if not found
-    public function get($which)
+    // Get all the airports in an array
+    public function allAsArray()
```

Comment:
The difference between this and the `all` function in the `Memory_Model` class is the `all` function returns the airports as an array of airport objects. 
You just have to "access" the fields of each airport object with `$airport->departsFrom` instead of `$airport['departsFrom']`

I don't think we need this function at all. See here for changes that will help remove this function: 755779c96c31ed65b230184e21f14efa25f1fabc

Reply:
ya i was wondering about that. thanks